src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -1670,10 +1712,7 @@ dataType
 NiftiImageData<dataType>::
 get_inner_product(const NiftiImageData &other) const
 {
-	dataType s;
-	this->dot(other, &s);
-	return s;
-//    return std::inner_product(this->begin(),this->end(),other.begin(),dataType(0));
+    return (dataType)this->dot(other);
```

Comment:
```suggestion
    return static_cast<dataType>(this->dot(other));
```
otherwise the brackets are rather confusing

Reply:
thanks, adopted